osu.Game/Overlays/FirstRunSetupOverlay.cs

```diff
@@ -0,0 +1,397 @@
+                        // The base call will handle dismissal of the overlay.
+                        break;
+                }
+            }
+
+            return base.OnPressed(e);
+        }
+
+        public override void Show()
+        {
+            // if we are valid for display, only do so after reaching the main menu.
+            performer.PerformFromScreen(screen =>
+            {
+                MainMenu menu = (MainMenu)screen;
+
+                // Eventually I'd like to replace this with a better method that doesn't access the screen.
+                // Either this dialog would be converted to its own screen, or at very least be "hosted" by a screen pushed to the main menu.
+                // Alternatively, another method of disabling notifications could be added to `INotificationOverlay`.
+                if (menu != null)
+                {
+                    overlayActivationMode = menu.OverlayActivationMode.GetBoundCopy();
+                    overlayActivationMode.Value = OverlayActivation.UserTriggered;
+                }
+
+                base.Show();
+            }, new[] { typeof(MainMenu) });
+        }
+
+        protected override void PopIn()
+        {
+            base.PopIn();
+
+            this.ScaleTo(scale_when_hidden)
+                .ScaleTo(1, 400, Easing.OutElasticHalf);
+
+            this.FadeIn(400, Easing.OutQuint);
+
+            if (currentStepIndex == null)
+                showFirstStep();
+        }
+
+        protected override void PopOut()
+        {
+            if (overlayActivationMode != null)
+            {
+                // If this is non-null we are guaranteed to have come from the main menu.
+                overlayActivationMode.Value = OverlayActivation.All;
+                overlayActivationMode = null;
+            }
+
+            if (currentStepIndex != null)
+            {
+                notificationOverlay.Post(new SimpleNotification
+                {
+                    Text = FirstRunSetupOverlayStrings.ClickToResumeFirstRunSetupAtAnyPoint,
+                    Icon = FontAwesome.Solid.Horse,
+                    Activated = () =>
+                    {
+                        Show();
+                        return true;
+                    },
+                });
+            }
+            else
+            {
+                stack?.FadeOut(100)
+                     .Expire();
+            }
+
+            base.PopOut();
+
+            this.ScaleTo(0.96f, 400, Easing.OutQuint);
+            this.FadeOut(200, Easing.OutQuint);
+        }
+
+        private void showFirstStep()
+        {
+            Debug.Assert(currentStepIndex == null);
+
+            stackContainer.Child = stack = new ScreenStack
+            {
+                RelativeSizeAxes = Axes.Both,
+            };
+
+            currentStepIndex = -1;
+            showNextStep();
+        }
+
+        private void showLastStep()
```

Comment:
Not sure if just me but when I read `showLastStep()` just after `showFirstStep()` I was expecting a jump forward to the last step, but this is actually more like `showPreviousStep()`?